problem-spring-web-autoconfigure/src/main/java/org/zalando/problem/spring/web/autoconfigure/JacksonModulesRegistratorWithoutMvcAutoConfig.java

```diff
@@ -0,0 +1,35 @@
+package org.zalando.problem.spring.web.autoconfigure;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.springframework.boot.autoconfigure.AutoConfigureBefore;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnBean;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnWebApplication;
+import org.springframework.boot.autoconfigure.web.servlet.WebMvcAutoConfiguration;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.http.converter.HttpMessageConverter;
+import org.springframework.http.converter.json.Jackson2ObjectMapperBuilder;
+import org.springframework.http.converter.json.MappingJackson2HttpMessageConverter;
+import org.springframework.web.servlet.config.annotation.WebMvcConfigurationSupport;
+import org.springframework.web.servlet.config.annotation.WebMvcConfigurer;
+import org.zalando.problem.ProblemModule;
+import org.zalando.problem.violations.ConstraintViolationProblemModule;
+
+import java.util.List;
+
+/**
+ * Registers jackson modules when using @EnableWebMvc, which deactivates WebMvc AutoConfig
+ */
+@Configuration
+@ConditionalOnWebApplication
+@ConditionalOnBean(WebMvcConfigurationSupport.class)
+@AutoConfigureBefore(WebMvcAutoConfiguration.class)
```

Comment:
Aren't those two mutually exclusive?

Reply:
you mean WebMvcConfigurationSupport and WebMvcAutoConfiguration? you are right in a way, but the point of having both is, if i don't have the `..Before` annotation, this class will ALWAYS match because if there's autoconfigure, `conditionOnBean` will match due to what the autoconfigure did, and if there's no autoconfigure, it will also match because you used `@EnableWebMvc`, so the idea is, activate this class only in the case of `enableWebMvc` but always before the potential autonconfigure

Reply:
Does `WebMvcAutoConfiguration` provide a `WebMvcConfigurationSupport`?

Reply:
correct.
`org.springframework.boot.autoconfigure.web.servlet.WebMvcAutoConfiguration.EnableWebMvcConfiguration` is the one